app/controllers/api/application_plans_controller.rb

```diff
@@ -1,59 +1,48 @@
 # frozen_string_literal: true
 
 class Api::ApplicationPlansController < Api::PlansBaseController
-  before_action :activate_sidebar_menu
-
   activate_menu :serviceadmin, :applications, :application_plans
   sublayout 'api/service'
 
-  def index
-    @new_plan = ApplicationPlan
-  end
-
-  def new
-    @plan = collection.build params[:application_plan]
-  end
+  helper_method :default_plan_select_data, :plans_table_data
+  delegate :default_plan_select_data, :plans_table_data, to: :presenter
 
-  def edit
-    @plan = collection.includes(:plan_metrics, :usage_limits, :pricing_rules, service: :top_level_metrics).find(params[:id])
-  end
+  alias application_plans plans
 
-  # class super metod which is Api::PlansBaseController#create
-  # to create plan same way as all plans
-  #
+  # rubocop:disable Lint/UselessMethodDefinition We need these, otherwise integration tests will fail
   def create
-    super(application_plan_params)
+    super
   end
 
   def update
-    super(application_plan_params)
+    super
   end
 
   def destroy
     super
   end
+  # rubocop:enable Lint/UselessMethodDefinition
 
   def masterize
-    assign_plan!(@service, :default_application_plan)
-    flash[:notice] = 'The default plan has been changed.'
-    redirect_to admin_service_application_plans_path(@service)
+    super(@service, :default_application_plan)
   end
 
   protected
 
+  def plan_type
+    :application_plan
+  end
+
+  # TODO: leave or delete? When is it current_account the scope?
```

Comment:
Yeah, can't find any view which would not be under the service scope either...

Reply:
I added a `raise unless @service` and there's a bunch of tests failing so apparently it is possible.